com.unity.mobile.notifications/Runtime/iOS/AuthorizationRequest.cs

```diff
@@ -68,26 +67,53 @@ internal struct iOSAuthorizationRequestData
     /// </example>
     public class AuthorizationRequest : IDisposable
```

Comment:
I didn't get why we need to lock `AuthorizationRequest` for getter...

Reply:
We set properties from other thread, so we have a race condition between read&write, which might not be safe for strings. Also, you may have race condition if you read multiple properties, putting locks place ensures all properties are set atomically by the writer before they are made available to the reader.